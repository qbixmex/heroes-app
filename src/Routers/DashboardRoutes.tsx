import { Route, Routes } from "react-router-dom";
import DCScreen from "../Components/DC/DCScreen";
import MarvelScreen from "../Components/Marvel/MarvelScreen";
import SearchScreen from "../Components/Search/SearchScreen";
import { Navbar } from "../Components/UI/Navbar";
import HeroScreen from '../Components/Hero/HeroScreen';

const DashboardRoutes = () => {
  return (
    <>
      <Navbar />

      <Routes>
        <Route path="marvel" element={ <MarvelScreen /> } />
        <Route path="dc" element={ <DCScreen /> } />
        <Route path="search" element={ <SearchScreen /> } />
        <Route path="hero" element={ <HeroScreen /> } />
        <Route path="/" element={ <MarvelScreen /> } />
      </Routes>
    </>
  );
};

export default DashboardRoutes;
