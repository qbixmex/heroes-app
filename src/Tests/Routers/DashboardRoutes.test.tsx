import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../Auth/authContext';
import DashboardRoutes from '../../Routers/DashboardRoutes';


//@ts-ignore
describe("Test on <DashboardRoutes />", () => {

  const contextValue = {
    user: {
      name: "John Doe",
      logged: true,
    },
    dispatch: jest.fn()
  };

  test("Should match with snapshot - Marvel", () => {

    const wrapper = mount(
      <AuthContext.Provider value={ contextValue }>
        <MemoryRouter initialEntries={ ["/"] }>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    const name = wrapper.find(".text-info").text().trim();
    const title = wrapper.find("#marvelTitle").text().trim();

    expect( wrapper ).toMatchSnapshot();
    expect( name ).toBe( contextValue.user.name );
    expect( title ).toBe("Marvel Screen");

  });

  test("Should match with snapshot - DC", () => {

    const wrapper = mount(
      <AuthContext.Provider value={ contextValue }>
        <MemoryRouter initialEntries={ ["/dc"] }>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    const name = wrapper.find(".text-info").text().trim();
    const title = wrapper.find("#dcTitle").text().trim();

    expect( wrapper ).toMatchSnapshot();
    expect( name ).toBe( contextValue.user.name );
    expect( title ).toBe("DC Screen");

  });
});
