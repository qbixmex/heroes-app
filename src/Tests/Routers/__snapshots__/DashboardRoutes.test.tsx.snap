// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <DashboardRoutes /> Should match with snapshot - DC 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark bg-gradient px-4 mb-4"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link "
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <button
            className="nav-link btn text-info"
          >
            John Doe
          </button>
        </li>
        <li
          className="nav-item"
        >
          <button
            className="nav-link btn"
            id="logout"
            onClick={[Function]}
          >
            Logout
          </button>
        </li>
      </ul>
    </div>
  </nav>,
  <section
    className="container"
  >
    Array [
      <h1
        className="text-center display-1 green"
        id="dcTitle"
      >
        DC Screen
      </h1>,
      <hr />,
      <div
        className="row rows-cols-1 animate__animated animate__fadeIn"
      >
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/dc-batman"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Batman"
                    className="img-fluid rounded"
                    src="/assets/dc-batman.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Batman
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/dc-superman"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Superman"
                    className="img-fluid rounded"
                    src="/assets/dc-superman.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Superman
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/dc-flash"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Flash"
                    className="img-fluid rounded"
                    src="/assets/dc-flash.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Flash
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/dc-green"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Green Lantern"
                    className="img-fluid rounded"
                    src="/assets/dc-green.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Green Lantern
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/dc-arrow"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Green Arrow"
                    className="img-fluid rounded"
                    src="/assets/dc-arrow.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Green Arrow
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/dc-wonder"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Wonder Woman"
                    className="img-fluid rounded"
                    src="/assets/dc-wonder.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Wonder Woman
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/dc-martian"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Martian Manhunter"
                    className="img-fluid rounded"
                    src="/assets/dc-martian.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Martian Manhunter
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/dc-robin"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Robin/Nightwing"
                    className="img-fluid rounded"
                    src="/assets/dc-robin.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Robin/Nightwing
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/dc-blue"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Blue Beetle"
                    className="img-fluid rounded"
                    src="/assets/dc-blue.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Blue Beetle
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/dc-black"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Black Canary"
                    className="img-fluid rounded"
                    src="/assets/dc-black.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Black Canary
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>,
    ]
  </section>,
]
`;

exports[`Test on <DashboardRoutes /> Should match with snapshot - Marvel 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark bg-gradient px-4 mb-4"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link "
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link "
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <button
            className="nav-link btn text-info"
          >
            John Doe
          </button>
        </li>
        <li
          className="nav-item"
        >
          <button
            className="nav-link btn"
            id="logout"
            onClick={[Function]}
          >
            Logout
          </button>
        </li>
      </ul>
    </div>
  </nav>,
  <section
    className="container"
  >
    Array [
      <h1
        className="text-center display-1 green"
        id="marvelTitle"
      >
        Marvel Screen
      </h1>,
      <hr />,
      <div
        className="row rows-cols-1 animate__animated animate__fadeIn"
      >
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-spider"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Spider Man"
                    className="img-fluid rounded"
                    src="/assets/marvel-spider.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Spider Man
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-captain"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Captain America"
                    className="img-fluid rounded"
                    src="/assets/marvel-captain.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Captain America
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-iron"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Iron Man"
                    className="img-fluid rounded"
                    src="/assets/marvel-iron.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Iron Man
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-thor"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Thor"
                    className="img-fluid rounded"
                    src="/assets/marvel-thor.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Thor
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-hulk"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Hulk"
                    className="img-fluid rounded"
                    src="/assets/marvel-hulk.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Hulk
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-wolverine"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Wolverine"
                    className="img-fluid rounded"
                    src="/assets/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Wolverine
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-daredevil"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Daredevil"
                    className="img-fluid rounded"
                    src="/assets/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Daredevil
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-hawkeye"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Hawkeye"
                    className="img-fluid rounded"
                    src="/assets/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Hawkeye
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-cyclops"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Cyclops"
                    className="img-fluid rounded"
                    src="/assets/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Cyclops
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-silver"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Silver Surfer"
                    className="img-fluid rounded"
                    src="/assets/marvel-silver.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Silver Surfer
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>,
    ]
  </section>,
]
`;
